src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php

```diff
@@ -510,7 +546,7 @@ private function doRead($sessionId)
             $sessionRows = $selectStmt->fetchAll(\PDO::FETCH_NUM);
 
             if ($sessionRows) {
-                if ($sessionRows[0][1] + $sessionRows[0][2] < time()) {
+                if ($sessionRows[0][1] < time() || (isset($sessionRows[0][2]) && $sessionRows[0][1] + $sessionRows[0][2] < time())) {
```

Comment:
I would base this check on a ternary check for `false === $this->lifetimeCol`, so that it's clearer and easier to spot that this needs to be changed in 4.0